src/de/mod.rs

```diff
@@ -1672,10 +1673,185 @@
 //!
 //! Instead of writing such functions manually, you also could try <https://lib.rs/crates/serde-query>.
 //!
+//! ## Tagged Enums
+//! Tagged enums are currently not supported because of an issue in the Serde design (See [#1183] and [#586]).
+//! This can be worked around by manually implementing deserialize with `#[serde(deserialize_with = "func")]
+//! or implenting Deserialize, but this can get very tedious very fast for files with large amounts of tagged
+//! enums. To decode this XML pattern:
+//!
+//! ```xml
+//! <root>
+//!     <one tag="spam">
+//!         <spam ham="0.0" />
+//!     </one>
+//!     <two tag="ham">
+//!         <ham />
+//!     <ham>
+//! </root>
+//!
+//! ```
+//! 
+//! Define these two macros:
```

Comment:
I think, it would be better to write something like "You could, for example, define these two macros", to emphasis that the solution is not general and contains some simplifications, which is also would be great to list here:
- the first attribute should be the name of tag of enum
- only newtype (ADT) and unit-style (ZST) variants are supported